### PR TITLE
feat: Support MFM quote message input

### PR DESCRIPTION
**Pull request by liamcho**
Screen Shot 2023-09-12 at 5 27 52 PM

### CODE REVIEW

File: src/ui/QuoteMessage/utils.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+import {match} from "ts-pattern";
+import {FileMessage, MultipleFilesMessage, UploadedFileInfo} from "@sendbird/chat/message";
+import {Thumbnail} from "@sendbird/chat/lib/__definition";
+
+export function getMessageFirstFileType(message: CoreMessageType): string {
+  return match(message)
+    .when(m => isFileMessage(m), () => {
+      return (message as FileMessage)?.type ?? '';
+    })
+    .when(m => isMultipleFilesMessage(m), () => {
```
Comment (AhyoungRyu) [inline src/ui/QuoteMessage/utils.ts:11]: nit; this works same. 

```suggestion
    .when(isFileMessage, () => {
      return (message as FileMessage)?.type ?? '';
    })
    .when(isMultipleFilesMessage, () => {
```

File: src/ui/QuoteMessage/utils.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      const castedMessage: MultipleFilesMessage = message as MultipleFilesMessage;
+      return getFirstFileInfo(castedMessage)?.mimeType ?? '';
+    })
+    .otherwise(() => {
+      return '';
+    });
+}
+
+function getFirstFileInfo(message: MultipleFilesMessage): UploadedFileInfo {
```
Comment (AhyoungRyu) [inline src/ui/QuoteMessage/utils.ts:20]:
```suggestion
function getFirstFileInfo(message: MultipleFilesMessage): UploadedFileInfo | null {
```
It seems it returns `null` too.

File: src/ui/QuoteMessage/utils.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+export function getMessageFirstFileName(message: CoreMessageType): string {
+  return match(message)
+    .when(m => isFileMessage(m), () => {
+      const castedMessage: FileMessage = message as FileMessage;
+      return castedMessage?.name ?? '';
+    })
+    .when(m => isMultipleFilesMessage(m), () => {
+      const castedMessage: MultipleFilesMessage = message as MultipleFilesMessage;
+      return getFirstFileInfo(castedMessage)?.fileName ?? '';
```
Comment (AhyoungRyu) [inline src/ui/QuoteMessage/utils.ts:33]:
```suggestion
      return (message as FileMessage)?.name ?? '';
    })
    .when(isMultipleFilesMessage, () => {
      return (message as MultipleFilesMessage)?.fileName ?? '';
```

Like you did at the above in `getMessageFirstFileType`,  we can make it simpler in that way. 
Same for `getMessageFirstFileUrl` or `getMessageFirstFileThumbnails`.

File: src/utils/index.ts [hunk 4]
```diff
@@ -239,6 +241,8 @@ export const isEnabledOGMessage = (message: UserMessage): boolean => (
 );
 
 export const getUIKitMessageTypes = (): UIKitMessageTypes => ({ ...UIKitMessageTypes });
+
+// Do not use this for MultipleFilesMessage. Use isMultipleFilesMessage() instead.
```
Comment (AhyoungRyu) [inline src/utils/index.ts:245]:
```suggestion
/**
 * Do not use this for MultipleFilesMessage. Use isMultipleFilesMessage() instead.
 */
```
nit;
To give a hint in IDE, the comment format should be like the above. 
Screenshot 2023-09-12 at 5 55 13 PM

File: src/ui/QuoteMessage/utils.ts (hunk 1 above)
Comment (liamcho) [reply src/ui/QuoteMessage/utils.ts:11]: oh nice, thanks!

File: src/ui/QuoteMessage/utils.ts (hunk 2 above)
Comment (liamcho) [reply src/ui/QuoteMessage/utils.ts:20]: thanks for the catch!

File: src/ui/QuoteMessage/utils.ts (hunk 3 above)
Comment (liamcho) [reply src/ui/QuoteMessage/utils.ts:33]: ah nice suggestion! castedMessage is no longer needed after replacing repeating code with `getFirstFileInfo`.

File: src/utils/index.ts (hunk 4 above)
Comment (liamcho) [reply src/utils/index.ts:245]: nice!

**Review by HoonBaek — APPROVED** [review]
LGTM

ps. Let's migrate the functions `is~~~Message` and the function `getUIKitMessageType` ASAP

**Pull request merged**